sklearn/preprocessing/_encoders.py

```diff
@@ -81,7 +81,7 @@ def _fit(self, X, handle_unknown='error'):
 
         if self._categories != 'auto':
             if len(self._categories) != n_features:
-                raise ValueError("Shape mismatch: if n_values is an array,"
+                raise ValueError("Shape mismatch: if categories is an array,"
```

Comment:
seems like there's no test for this one, could you please add a test for it?

Reply:
@adrinjalali Added. Please check.

Reply:
pep8 issues.

Reply:
could you please clarify? are you referring to the same point as made by jnothman or anything specific like naming conventions on the test..